internal/services/eventgrid/eventgrid_partner_namespace_resource.go

```diff
@@ -0,0 +1,331 @@
+	}
+
+	ipRules := make([]partnernamespaces.InboundIPRule, 0)
+	for _, v := range input {
+		ipRules = append(ipRules, partnernamespaces.InboundIPRule{
+			Action: pointer.To(partnernamespaces.IPActionType(v.Action)),
+			IPMask: pointer.To(v.IpMask),
+		})
+	}
+	return &ipRules
+}
+
+func flattenPartnerInboundIPRules(ipRules *[]partnernamespaces.InboundIPRule) []PartnerInboundIpRuleModel {
+	output := make([]PartnerInboundIpRuleModel, 0)
+
+	if ipRules == nil || len(*ipRules) == 0 {
+		return output
+	}
+
+	for _, v := range *ipRules {
+		output = append(output, PartnerInboundIpRuleModel{
+			IpMask: pointer.From(v.IPMask),
+			Action: string(pointer.From(v.Action)),
+		})
+	}
+	return output
+}
+
+func (r EventGridPartnerNamespaceResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.PartnerNamespaces
+
+			id, err := partnernamespaces.ParsePartnerNamespaceID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var config EventGridPartnerNamespaceResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if existing.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil", *id)
+			}
+
+			if existing.Model.Properties == nil {
+				return fmt.Errorf("retrieving %s: `model.Properties` was nil", *id)
+			}
+
+			model := existing.Model
+
+			if metadata.ResourceData.HasChange("local_auth_enabled") {
+				model.Properties.DisableLocalAuth = pointer.To(!config.LocalAuthEnabled)
+			}
+			if metadata.ResourceData.HasChange("inbound_ip_rule") {
+				model.Properties.InboundIPRules = expandPartnerInboundIPRules(config.InboundIPRules)
+			}
+			if metadata.ResourceData.HasChange("public_network_access_enabled") {
+				if config.PublicNetworkAccessEnabled {
+					model.Properties.PublicNetworkAccess = pointer.To(partnernamespaces.PublicNetworkAccessEnabled)
+				} else {
+					model.Properties.PublicNetworkAccess = pointer.To(partnernamespaces.PublicNetworkAccessDisabled)
+				}
+			}
+			if metadata.ResourceData.HasChange("tags") {
+				model.Tags = pointer.To(config.Tags)
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, *model); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r EventGridPartnerNamespaceResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.PartnerNamespaces
+
+			id, err := partnernamespaces.ParsePartnerNamespaceID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("parsing %q: %+v", metadata.ResourceData.Id(), err)
```

Comment:
Since the `Parse*` functions already return a `parsing <input>` error we don't need to wrap it and can just return the `err` variable

```suggestion
				return err
```